Fla 130 ability to remove blobs

sondre81 opened the pull request:
Added ability to remove storage blobs after a given time set in the crd.
Wrote unit tests.

sondre81 committed:
WIP | attempting to add lifecycle rules

sondre81 committed:
WIP | Added tests and removed AzureConfig from BCS

sondre81 committed:
Removed commented code

sondre81 committed:
Added test if SA is empty and if not

sondre81 committed:
Added test for BCS delete

sondre81 committed:
Added lifespanDays to CRD

sondre81 committed:
Added lifespanDays to crd, refactored test to suit changes.

sondre81 committed:
Added SR test

sondre81 commented:
Bør kansje endre Deploymenten til å kun gjelde Alpha til å begynne med? Greit å sjekke at ting ikke krasjer før man kaster det ut i Beta/Api.

murillio4 commented on src/main/java/no/fintlabs/azure/TagNames.java:
```diff
@@ -10,6 +10,7 @@ public class TagNames {
     public static final String TAG_PART_OF = "part-of";
     public static final String TAG_CRD_NAME = "crd-name";
     public static final String TAG_CRD_NAMESPACE = "crd-namespace";
+    public static final String TAG_LIFESPAN = "60L";
```
Burde taggen hete `60L`? Eller er det mening at standard verdien er `60` og så skal taggen hete noe annet?

murillio4 commented on src/main/java/no/fintlabs/azure/storage/StorageAccountService.java:
```diff
@@ -57,6 +57,7 @@ public StorageAccount add(FlaisCrd<? extends AzureSpec> crd, String path, Storag
                 .withTag(TAG_INSTANCE, crd.getMetadata().getLabels().get("app.kubernetes.io/instance"))
                 .withTag(TAG_PART_OF, crd.getMetadata().getLabels().get("app.kubernetes.io/part-of"))
                 .withTag(TAG_CRD_NAME, crd.getMetadata().getName())
+                .withTag(TAG_LIFESPAN, "60L")
```
Burde denne være satt til `lifespanDays`?

murillio4 commented on src/main/java/no/fintlabs/azure/storage/StorageResource.java:
```diff
@@ -40,12 +41,15 @@ public static StorageResource desired() {
 
     public static StorageResource of(StorageAccount storageAccount) {
         StorageType storageType = StorageType.valueOf(storageAccount.tags().getOrDefault(TAG_TYPE, StorageType.UNKNOWN.name()));
+        long lifespanDays = Long.parseLong(storageAccount.tags().getOrDefault(TAG_LIFESPAN, "60L"));
```
`60L` burde vært trukket ut som en konstant.

murillio4 reviewed (changes requested):
Hvis du får ryddet opp i dette, ser på kommentaren min om navn på lifespan-labelen, legger til validering av minimum og maksimum for lifespanDays, og tester dette, så er dette på god vei til å bli bra! :)

sondre81 committed:
Made changes from feedback on pull request

murillio4 commented on src/main/java/no/fintlabs/azure/storage/StorageAccountService.java:
```diff
@@ -57,6 +58,7 @@ public StorageAccount add(FlaisCrd<? extends AzureSpec> crd, String path, Storag
                 .withTag(TAG_INSTANCE, crd.getMetadata().getLabels().get("app.kubernetes.io/instance"))
                 .withTag(TAG_PART_OF, crd.getMetadata().getLabels().get("app.kubernetes.io/part-of"))
                 .withTag(TAG_CRD_NAME, crd.getMetadata().getName())
+                .withTag(TAG_LIFESPAN_DAYS, String.valueOf(lifespanDays))
```
Er det snakk om lifespan på storage accounten, eller lifespan på blobs i storage accounten?

Blob storage støtter lifespan/retention på blobs, men om man ønsker å sette denne til en custom verdi kan ikke dette gjøres gjennom tags. Det må lages en ny lifecycle rule hvor man setter lifespan på blobs.

Tenker det er fint å bevare lifespan tagen på storage accounten slik at det blir lettere for oss å søke, etc...

https://learn.microsoft.com/en-us/azure/storage/blobs/lifecycle-management-policy-configure?tabs=azure-portal

murillio4 replied:
Et annet spm er om man alltid vil ha lifespan? Skal den være optional? Så om den ikke er satt vil vi ikke opprette en lifespan regel.

murillio4 commented on src/test/java/no/fintlabs/azure/storage/blob/BlobContainerServiceTest.java:
Denne må vi rydde litt opp i. Jeg tenker vi bruker [Azurite](https://github.com/Azure/Azurite) for dette. Jeg kan sette opp

murillio4 committed:
feat: Add tests for BlobContainerService and StorageAccountService

- Changed reconciliation loop for BlobContainerReconciler and FileShareReconciler to set storage account name in annotations from workflow context.
- Blob storage lifetime not set if no time provided
